src/modules/margin_tokens/LeveragedToken.ts

```diff
@@ -140,6 +147,13 @@ export default class LeveragedToken extends MarginToken {
     const position: Position = await this.margin.getPosition(positionId);
     const loanOffering: SignedLoanOffering = this.prepareMintLoanOffering(position);
 
+    const collateral: BigNumber = await this.margin.getPositionBalance(positionId);
```

Comment:
So I can't comment on the line, but you should fix this for `#mint` as well. Right now it just passes in `tokensToMint` as the amount of principal to add to the position. You need to do the same partial amount thing there too

Reply:
migrated over to private function `calculatePrincipal`